packages/webpack-cli/src/webpack-cli.ts

```diff
@@ -756,18 +756,22 @@ class WebpackCLI implements IWebpackCLI {
           value: string,
           previous: Record<string, BasicPrimitive | object> = {},
         ): Record<string, BasicPrimitive | object> => {
-          // for https://github.com/webpack/webpack-cli/issues/2642
-          if (value.endsWith("=")) {
-            value.concat('""');
-          }
-
           // This ensures we're only splitting by the first `=`
           const [allKeys, val] = value.split(/=(.+)/, 2);
           const splitKeys = allKeys.split(/\.(?!$)/);
 
           let prevRef = previous;
 
           splitKeys.forEach((someKey, index) => {
+            // https://github.com/webpack/webpack-cli/issues/3284
+            if (someKey.endsWith("=")) {
+              // remove '=' from key
+              someKey = someKey.slice(0, -1);
+              // @ts-expect-error we explicitly want to set it to undefined
+              prevRef[someKey] = undefined;
```

Comment:
I don't think we might want to set `TEST1=""` to be `TEST1: undefined`? The value exists as empty string? It is not left undefined like other cases? thoughts @alexander-akait?

Reply:
`TEST1=""` is not undefined. `TEST1=` see test cases.

Reply:
https://github.com/webpack/webpack-cli/pull/3286/files#diff-e4aa9d12f691761ac121306d5d8b35208d9438d28ab8c1e093ac14703390e673R129

Reply:
<img width="578" alt="Screenshot 2022-06-10 at 9 36 12 AM" src="https://user-images.githubusercontent.com/32865581/172988312-1aae441c-c9df-4fb6-a60c-f01fec686c09.png">

Any Idea on what is happening? I ran it in this branch only?  `TEST1=""` and the env logs TEST1 to be undefined although it was set to be empty string?

Reply:
@harish-sethuraman In the shell, you need to escape `"` character.

`npx webpack --env TEST=\"\"`

[See this test case](https://github.com/webpack/webpack-cli/pull/3286/files#diff-e4aa9d12f691761ac121306d5d8b35208d9438d28ab8c1e093ac14703390e673R166-R176)

Reply:
Yea If we escape it I get an empty string within a string `TEST1: '""'` which is expected since we escape. But for cases where we don't (TEST="") Consider if Im doing checks with length like `env.TEST1.length === 0` it is going to fail since there it is undefined. Im not sure if this is the expected but I feel we could improve here.

This is what 
When I set the `VITE_SOME_KEY=""` this is what I get in the env object but in webpack it is undefined
<img width="121" alt="Screenshot 2022-06-10 at 9 51 28 AM" src="https://user-images.githubusercontent.com/32865581/172989755-df6150e3-5918-43ac-91c6-52cf9426225f.png">

Even when I ran commands like TEST="" in the cli and logged the process.env I see TEST: '' and not TEST: undefined


Feel free to resolve if that is the expected behaviour